api/src/main/java/org/eclipse/microprofile/metrics/MetricID.java

```diff
@@ -124,7 +124,7 @@ public MetricID(String name, Tag... tags) {
         Optional<String> applicationName = config.getOptionalValue(APPLICATION_NAME_VARIABLE, String.class);
         applicationName.ifPresent(appName -> {
             if(!appName.isEmpty()) {
-                addTag(new Tag(APPLICATION_NAME_TAG, appName));
+                this.tags.put(APPLICATION_NAME_TAG, appName);
```

Comment:
perhaps a bit picky... but for consistency it would be better to create an addTag(String name, String value) method that the existing addTag(Tag) method and the code above could call.

Reply:
I had this idea too, but it just felt as an unnecessary method call and four lines of code instead of one; but if you insist we can do that